### PR TITLE
JFB-12. Spring Boot1.2

### PR DESCRIPTION
**Pull request by Sialenius**

### CODE REVIEW

File: src/main/java/com/project/jfb/config/ThisIsMyFirstConditionalBean.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@Configuration
+@ComponentScan
+public class ThisIsMyFirstConditionalBean {
+
+    @Bean
+    @ConditionalOnProperty(name = "isConditionalBeanAvailable", havingValue = "true")
+    public String useMyConditionalBean() {
+
+        return "Conditional bean is available";
+    }
```
Comment (AtticSiol) [inline src/main/java/com/project/jfb/config/ThisIsMyFirstConditionalBean.java]: when you create bean, then you need not just create a method with some condition, you also need to return this been, not a `String`:
```java
    @Bean
    @ConditionalOnProperty(name = "isConditionalBeanAvailable", havingValue = "true")
    public ThisIsMyFirstConditionalBean thisIsMyFirstConditionalBean() {
        return new ThisIsMyFirstConditionalBean();
    }
```

File: src/main/java/com/project/jfb/controller/TicketController.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("tickets") //http://localhost:8081/tickets
+public class TicketController {
+
+    @Autowired
+    TicketService ticketService;
```
Comment (AtticSiol) [inline src/main/java/com/project/jfb/controller/TicketController.java]: 1. inject beans with the help of constructor injection, not with field injection
2. spring beans have to be `private final`

File: src/main/java/com/project/jfb/controller/TicketController.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @GetMapping
+    public List<TicketRest> getAllTickets() {
+        List<TicketRest> returnValue = new ArrayList<>();
+
+        for (TicketDto t: ticketService.getAllTickets()) {
+            TicketRest temptTicketRest = new TicketRest();
+            BeanUtils.copyProperties(t, temptTicketRest);
+            returnValue.add(temptTicketRest);
```
Comment (AtticSiol) [inline src/main/java/com/project/jfb/controller/TicketController.java]: this logic has to be on service layer

File: src/main/java/com/project/jfb/controller/TicketController.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("tickets") //http://localhost:8081/tickets
```
Comment (AtticSiol) [inline src/main/java/com/project/jfb/controller/TicketController.java]: not `"tickets"`, use `"/tickets"`

File: src/main/java/com/project/jfb/controller/TicketController.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+        TicketDto createTicket = ticketService.saveTicket(ticketDto);
+
+        TicketRest returnValue = new TicketRest();
+        BeanUtils.copyProperties(createTicket, returnValue);
+
+        return returnValue;
+    }
+
+    @PutMapping("/{ticketId}/update-type")
```
Comment (AtticSiol) [inline src/main/java/com/project/jfb/controller/TicketController.java]: `put` already means update, use just `"/{ticketId}/type"`

File: src/main/java/com/project/jfb/service/TicketService.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class TicketService {
+
+    @Autowired
+    TicketRepository ticketRepository;
```
Comment (AtticSiol) [inline src/main/java/com/project/jfb/service/TicketService.java]: the same

File: src/main/java/com/project/jfb/config/ThisIsMyFirstConditionalBean.java (hunk 1 above)
Comment (Sialenius) [reply src/main/java/com/project/jfb/config/ThisIsMyFirstConditionalBean.java]: Fixed

File: src/main/java/com/project/jfb/controller/TicketController.java (hunk 2 above)
Comment (Sialenius) [reply src/main/java/com/project/jfb/controller/TicketController.java]: Fixed

File: src/main/java/com/project/jfb/controller/TicketController.java (hunk 4 above)
Comment (Sialenius) [reply src/main/java/com/project/jfb/controller/TicketController.java]: Fixed

File: src/main/java/com/project/jfb/controller/TicketController.java (hunk 5 above)
Comment (Sialenius) [reply src/main/java/com/project/jfb/controller/TicketController.java]: Fixed. Thank you

File: src/main/java/com/project/jfb/service/TicketService.java (hunk 6 above)
Comment (Sialenius) [reply src/main/java/com/project/jfb/service/TicketService.java]: Fixed

File: src/main/java/com/project/jfb/controller/TicketController.java (hunk 3 above)
Comment (Sialenius) [reply src/main/java/com/project/jfb/controller/TicketController.java]: Fixed

**Pull request merged**